language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/workspace/BallerinaWorkspaceManager.java

```diff
@@ -154,6 +160,34 @@ public Optional<Project> project(Path filePath) {
         return projectPair(projectRoot(filePath)).map(ProjectPair::project);
     }
 
+    /**
+     * Loads the project from the path provided.
+     *
+     * @param filePath ballerina project or standalone file path
+     * @return project of applicable type
+     */
+    @Override
+    public Project loadProject(Path filePath) throws ProjectException, WorkspaceDocumentException, EventSyncException {
+        Project project;
+        Optional<Project> optionalProject = project(ProjectPaths.packageRoot(filePath));
+        if (optionalProject.isPresent()) {
+            project = optionalProject.get();
+        } else {
+            project = createOrGetProjectPair(filePath, LSContextOperation.LOAD_PROJECT.getName()).project();
+        }
```

Comment:
Here, can't we simplify the logic as follows instead of doing an additional `project(ProjectPaths.packageRoot(filePath))` call, cuz it seems we already do the intended validation within the  `createOrGetProjectPair` method itself?

```suggestion
        Project project = createOrGetProjectPair(filePath, LSContextOperation.LOAD_PROJECT.getName()).project();
```

Reply:
Here, we should publish diagnostics only if it was created new. @gayalkuruppu we should update the logic to publish diagnostics only if the project was loaded newly.

Reply:
@NipunaRanasinghe Agree but to address https://github.com/ballerina-platform/ballerina-lang/pull/38032#discussion_r991956784 kept the if else statement in [2151f4b](https://github.com/ballerina-platform/ballerina-lang/pull/38032/commits/2151f4b1f417ff2cf2e9f37b189e61035bf764b1)